### PR TITLE
Postprocessor for executing shell commands.

### PR DESCRIPTION
**Pull request by jperryhouts**
I think this is handy. I have a python script for generating simple figures based on the output, and this plugin allows me to automatically run that script at the end of each timestep. It could also be used for all kinds of other stuff (progress notifications, etc)

One question: I remember there being a mechanism for forcing the order of plugins. Is that the case, and how could I use it to force this plugin to run last?

### CODE REVIEW

File: source/postprocess/command.cc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      if (on_all_processes ||
+          (Utilities::MPI::this_mpi_process(this->get_mpi_communicator()) == 0))
+        {
+          // Check if a command-processor is available by calling system() with a
+          // null pointer. System is guaranteed to return non-zero if it finds
+          // a terminal and zero if there is none (like on the compute nodes of
+          // some cluster architectures, e.g. IBM BlueGene/Q)
+          AssertThrow(system((char *)0) != 0,
+                      ExcMessage("The \"command\" postprocessor required a command-processor, "
+                                 "which appears to be unavailable on this system."));
```
Comment (bangerth) [inline source/postprocess/command.cc]: I see you've been paying attention to recent mails :-)

File: source/postprocess/command.cc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+          const int error = system(command.c_str());
+
+          if (error != 0)
+            {
+              if (terminate_on_failure)
+                {
+                  AssertThrow(false, ExcMessage("Could not execute command <" +
+                                                command +
+                                                ">. Terminating process."));
```
Comment (bangerth) [inline source/postprocess/command.cc]: Could you also output the error code, e.g., via `Utilities::to_string(error)`?

File: source/postprocess/command.cc (hunk 1 above)
Comment (jperryhouts) [inline source/postprocess/command.cc]: Enough to know where to look for examples :)

**Comment by bangerth** [comment]
So I think this is an interesting postprocessor. I like the direction. We'll have to remember to disable it if/when we ever offer running ASPECT as a service on someone else's computer since it creates a security hole.

As for being able to tell where in the queue things run: I don't think that we can do that right now. I believe that we create postprocessors in the order in which they're listed in the input file, and that we run them in this order. It may be worth checking that this is true, and documenting in the blurb that describes the postprocessor that it will be run in this order.

**Comment by tjhei** [comment]
Can you add a simple test that runs something like ``/bin/bash -c "echo 'hi'"``?

/run-tests

**Comment by tjhei** [comment]
>  since it creates a security hole.

args, like my tester? :-)

**Comment by bangerth** [comment]
Exactly.

**Comment by bangerth** [comment]
Ah, once again, I forgot that Timo's tester only uses deal.II 8.2, which did not have `Utilities::to_string`. Can you make do with `Utilities::int_to_string` (which only takes unsigned integers, so you may have to add the negative sign by hand, if necessary).

**Comment by jperryhouts** [comment]
It's definitely a big security vulnerability if you're running an "untrusted" .prm. Perhaps it should be possible to disable it via compiler flag?

**Comment by jperryhouts** [comment]
Also, just based on one check, it looks like the postprocessors are in fact run in the order listed (assuming the outputs are printed in the order run), but I don't know that this is guaranteed to be true.

**Comment by bangerth** [comment]
Looks good to me if you undo the addition of the `parameters.prm` and `parameters.tex` files for the test.

**Comment by jperryhouts** [comment]
Done.

**Comment by tjhei** [comment]
please remove screen-output.tmp

**Comment by jperryhouts** [comment]
roger that

**Comment by bangerth** [comment]
Since @tjhei is running the tester, I'll leave this up to him. I'm ok with the patch.

@jperryhouts -- I keep forgetting it, but can you please also add an entry to changes.h? My apologies for the many go-arounds.

**Comment by jperryhouts** [comment]
OK. All good now, I think.

**Pull request merged**